pkg/sql/storageparam/tablestorageparam/table_storage_param.go

```diff
@@ -226,6 +226,28 @@ var tableParams = map[string]tableParam{
 			)
 		},
 	},
+	`ttl_expiration_expression`: {
+		onSet: func(ctx context.Context, po *Setter, semaCtx *tree.SemaContext, evalCtx *eval.Context, key string, datum tree.Datum) error {
+			stringVal, err := paramparse.DatumAsString(evalCtx, key, datum)
+			if err != nil {
+				return err
+			}
+			if po.tableDesc.RowLevelTTL == nil {
+				po.tableDesc.RowLevelTTL = &catpb.RowLevelTTL{}
+			}
+			po.tableDesc.RowLevelTTL.ExpirationExpr = catpb.Expression(stringVal)
+			return nil
+		},
+		onReset: func(po *Setter, evalCtx *eval.Context, key string) error {
+			return errors.WithHintf(
+				pgerror.Newf(
+					pgcode.InvalidParameterValue,
+					`resetting "ttl_expiration_expression" is not permitted`,
```

Comment:
this is technically permitted now that `crdb_internal_expiration` is set.

Reply:
Updated.